numpy/_typing/_add_docstring.py

```diff
@@ -149,4 +149,45 @@ def _parse_docstrings() -> str:
 
     """)
 
+add_newdoc('Array', repr(Array),
+    """
+    A `np.ndarray[tuple[*Ts], np.dtype[+ScalarType]] <numpy.ndarray>` type
+    alias :term:`generic <generic type>` w.r.t. its shape and
+    `dtype.type <numpy.dtype.type>`.
+
+    Can be used during runtime for typing arrays with a given dtype
+    and specified shape.  Particularly useful for functions which modify number
+    of axes.  If it is important to enforce integer axis sizes, use
+    `np.ndarray <numpy.ndarray>` for typing.  If axes dimensions are
+    variable, it is best to use `npt.NDArray <numpy.typing.NDArray>`.
```

Comment:
```suggestion
    unknown and/or irrelevant, it is best to use `npt.NDArray <numpy.typing.NDArray>`.
```
This doesn't sound quite right: axis variability is by no means guaranteed to be a problem here, as typevars/typevartuples can keep track of those just fine (in theory). It's really when the array shape is either unknown or (completely) irrelevant that there's no reason _not_ to use `NDArray`.

Reply:
Doing this, but is it worth leaving any hint that practice hasn't necessarily caught up to theory yet? (mypy still doesn't like `Array[*Shape, dtype]`, but it looks like pyright now accepts this)